packages/eas-cli/src/analytics/AnalyticsManager.ts

```diff
@@ -0,0 +1,230 @@
+  GATHER_CREDENTIALS_FAIL = 'submit cli gather credentials fail',
+  GATHER_ARCHIVE_ATTEMPT = 'submit cli gather archive attempt',
+  GATHER_ARCHIVE_SUCCESS = 'submit cli gather archive success',
+  GATHER_ARCHIVE_FAIL = 'submit cli gather archive fail',
+  SUBMIT_REQUEST_ATTEMPT = 'submit cli request attempt',
+  SUBMIT_REQUEST_SUCCESS = 'submit cli request success',
+  SUBMIT_REQUEST_FAIL = 'submit cli request fail',
+  API_KEY_DOWNLOAD_FAIL = 'submit cli credentials api key download fail',
+  API_KEY_DOWNLOAD_RETRY = 'submit cli credentials api key download fail temporary',
+  API_KEY_DOWNLOAD_SUCCESS = 'submit cli credentials api key download succeed',
+}
+
+export enum BuildEvent {
+  BUILD_COMMAND = 'build cli build command',
+  PROJECT_UPLOAD_ATTEMPT = 'build cli project upload attempt',
+  PROJECT_UPLOAD_SUCCESS = 'build cli project upload success',
+  PROJECT_UPLOAD_FAIL = 'build cli project upload fail',
+  GATHER_CREDENTIALS_ATTEMPT = 'build cli gather credentials attempt',
+  GATHER_CREDENTIALS_SUCCESS = 'build cli gather credentials success',
+  GATHER_CREDENTIALS_FAIL = 'build cli gather credentials fail',
+  CONFIGURE_PROJECT_ATTEMPT = 'build cli configure project attempt',
+  CONFIGURE_PROJECT_SUCCESS = 'build cli configure project success',
+  CONFIGURE_PROJECT_FAIL = 'build cli configure project fail',
+  BUILD_REQUEST_ATTEMPT = 'build cli build request attempt',
+  BUILD_REQUEST_SUCCESS = 'build cli build request success',
+  BUILD_REQUEST_FAIL = 'build cli build request fail',
+
+  BUILD_STATUS_COMMAND = 'build cli build status',
+
+  CREDENTIALS_SYNC_COMMAND = 'build cli credentials sync command',
+  CREDENTIALS_SYNC_UPDATE_LOCAL_ATTEMPT = 'build cli credentials sync update local attempt',
+  CREDENTIALS_SYNC_UPDATE_LOCAL_SUCCESS = 'build cli credentials sync update local success',
+  CREDENTIALS_SYNC_UPDATE_LOCAL_FAIL = 'build cli credentials sync update local fail',
+  CREDENTIALS_SYNC_UPDATE_REMOTE_ATTEMPT = 'build cli credentials sync update remote attempt',
+  CREDENTIALS_SYNC_UPDATE_REMOTE_SUCCESS = 'build cli credentials sync update remote success',
+  CREDENTIALS_SYNC_UPDATE_REMOTE_FAIL = 'build cli credentials sync update remote fail',
+
+  ANDROID_KEYSTORE_CREATE = 'build cli credentials keystore create',
+}
+
+export enum MetadataEvent {
+  APPLE_METADATA_DOWNLOAD = 'metadata cli download apple response',
+  APPLE_METADATA_UPLOAD = 'metadata cli upload apple response',
+}
+
+/**
+ * The interface for commands to use to log events to analytics.
+ */
+export interface IAnalyticsManager {
+  logEvent(name: AnalyticsEvent, properties: Record<string, any>): void;
+}
+
+/**
+ * The interface for commands to use to orchestrate the analytics system. Should only be necessary to use
+ * this within EASCommand.
+ */
+export interface IAnalayticsManagerWithOrchestration extends IAnalyticsManager {
+  setActor(actor: Actor): void;
+  flushAsync(): Promise<void>;
+}
```

Comment:
I'd prefer not to use this convention of starting the interface name with `I`. 

1. At the moment, it's not forbidden by ESLint but it should be.
2. We don't use this convention elsewhere in the CLI.